lib/streamlit/DeltaGenerator.py

```diff
@@ -1777,6 +1801,140 @@ def text_input(self, element, label, value=""):
         current_value = ui_value if ui_value is not None else value
         return str(current_value)
 
+    @_with_element
+    def number_input(
+        self,
+        element,
+        label,
+        value=NoValue(),
+        min_value=None,
+        max_value=None,
+        step=None,
+        format=None
+    ):
+        """Display a single-line numeric input widget.
+
+        Parameters
+        ----------
+        label : str or None
+            A short label explaining to the user what this input is for.
+        value : int/float or None
+            The value of this widget when it first renders.
+            default: min_value or 0
+        min_value : int/float or None
+            The minimum permitted value.
+        max_value : int/float or None
+            The maximum permitted value.
+        step : int/float or None
+            The stepping interval.
+            Defaults to 1 if the value is an int, 0.01 otherwise.
+            If the value is not specified, the format parameter will be used.
+        format : str or None
+            Printf/Python format string.
+
+
+        Returns
+        -------
+        int/float
+            The current value of the numeric input widget.
+
+        Example
+        -------
+        >>> number = st.number_input('Insert a number')
+        >>> st.write('The current number is ', number)
+
+        """
+
+        if isinstance(value, NoValue):
+            if min_value:
+                value = min_value
+            else:
+                value = 0
+
+        value = 0 if isinstance(value, NoValue) else value
+        int_value = isinstance(value, int)
+        float_value = isinstance(value, float)
+
+        if value is None:
+            raise ValueError("The value should either be an int/float")
+        else:
+            if format is None:
+                format = "%d" if int_value else "%0.2f"
+
+            if step is None:
+                if format is None:
```

Comment:
I think it's not possible for format to be None here since we set it above on line 1862

Reply:
you are right, done